docs/types.rst

```diff
@@ -902,6 +908,23 @@ possible:
 It is planned to remove this restriction in the future but currently creates
 some complications because of how arrays are passed in the ABI.
 
+.. index:: !inline;arrays
+
+Inline Arrays
+^^^^^^^^^^^^^
+
+You can initialise a statically sized memory array inline using syntax such as ``string[] memory myarray = new string[](4);``. For example::
+
+     pragma solidity ^0.4.16;
+     contract C {
+        function f() public pure returns (uint8[5]) {
+            string[4] memory adaArr = ["This", "is", "an", "array"];
+            return [1, 2, 3, 4, 5];
+        }
+    }
+
+ You can create and initialise multi-dimensional arrays in the same way, but filling arrays can cost a lot of gas, so make sure you optimise how you use them.
```

Comment:
I'm not sure about the spelling. Do we prefer ``optimise`` over ``optimize``? Right now we have 2 occurrences of ``optimize`` (one being the actual commandline option) and 1 occurrence of ``optimise`` in the documentation. :wink:
I think we should at least be consistent.

Reply:
@bit-shift We are trying to standardise to British English, but there are definitely still inconsistencies in the docs that we are slowly fixing. But what command line option are you referring to? That's not in this doc is it?